runtime-modules/content/src/vnft_auction/types.rs

```diff
@@ -0,0 +1,279 @@
+        ensure!(self.owner.eq(owner), Error::<T>::DoesNotOwnVNFT);
+        Ok(())
+    }
+}
+
+/// Owned vNFT representation
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct OwnedNFT<
+    AccountId,
+    Moment: BaseArithmetic + Copy,
+    CuratorGroupId: Default + Copy,
+    CuratorId: Default + Copy,
+    MemberId: Default + Copy,
+    Balance,
+> {
+    pub owner: AccountId,
+    pub transactional_status:
+        TransactionalStatus<AccountId, Moment, CuratorGroupId, CuratorId, MemberId, Balance>,
+    pub creator_royalty: Option<(AccountId, Royalty)>,
+}
+
+impl<
+        AccountId: PartialEq,
+        Moment: BaseArithmetic + Copy,
+        CuratorGroupId: Default + Copy,
+        CuratorId: Default + Copy,
+        MemberId: Default + Copy,
+        Balance,
+    > OwnedNFT<AccountId, Moment, CuratorGroupId, CuratorId, MemberId, Balance>
+{
+    pub fn is_owner(&self, account_id: &AccountId) -> bool {
+        self.owner.eq(account_id)
+    }
+}
+
+/// Enum, representing nft issuance status
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum NFTStatus<
+    AccountId,
+    Moment: BaseArithmetic + Copy,
+    CuratorGroupId: Default + Copy,
+    CuratorId: Default + Copy,
+    MemberId: Default + Copy,
+    Balance,
+> {
+    NoneIssued,
+    Owned(OwnedNFT<AccountId, Moment, CuratorGroupId, CuratorId, MemberId, Balance>),
+}
+
+impl<
+        AccountId,
+        Moment: BaseArithmetic + Copy,
+        CuratorGroupId: Default + Copy,
+        CuratorId: Default + Copy,
+        MemberId: Default + Copy,
+        Balance,
+    > Default for NFTStatus<AccountId, Moment, CuratorGroupId, CuratorId, MemberId, Balance>
+{
+    fn default() -> Self {
+        Self::NoneIssued
+    }
+}
+
+/// Either new auction, which requires vNFT issance or auction for already existing nft.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum AuctionMode {
+    // Auction, where nft issued at the end
+    WithIssuance(Option<Royalty>, Metadata),
+    // Auction for already existing nft
+    WithoutIsuance,
+}
+
+impl Default for AuctionMode {
+    fn default() -> Self {
+        Self::WithoutIsuance
+    }
+}
+
+/// Information on the auction being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct AuctionRecord<
+    AccountId,
+    Moment: BaseArithmetic + Copy,
+    CuratorGroupId: Default + Copy,
+    CuratorId: Default + Copy,
+    MemberId: Default + Copy,
+    Balance,
+> {
+    pub auctioneer: ContentActor<CuratorGroupId, CuratorId, MemberId>,
+    pub auctioneer_account_id: AccountId,
+    pub auction_mode: AuctionMode,
+    pub starting_price: Balance,
+    pub buy_now_price: Option<Balance>,
```

Comment:
Thinking about it, I don't think it makes sense to have both an open auction and a buy now price at the same time, because the buy now effectively caps the auction process, so I think we should just lift the buy now state into `TransactionalStatus`, for example `TransactionalStatus::BuyNow(price)`.

Reply:
Sorry, but i don't understand this thesis. Why doesn't it make sense to have a `buy_now_price` for open auction? 

From my perspective, auction with `buy_now_price` enabled, runs just like a regular auction, but at any time bidder have an ability to make a bid, which is equal or greater then `buy_now_price`. Afterwards he immediately become an auction winner at the cost of paying `buy_now_price`.

Reply:
I see what you mean, it oculd make sense to have both together, but it does run the risk of the creator loosing out from bidding that would have exceeded the buy now price, but ok. However, I think one should definitively also be able to sell an NFT even when there is no auction ongoing, this is important when there is perhaps not a vibrant and liquid market and attention for a given NFT, or when you want some price discovery. This is not possible now?

It could even actually be add to this the ability for people to submit bids which are lower than the buy now price, so that the issuer gains some information about what people are willing to pay. If the issuer set the buy now price to the latest bid, then the transaction would happen. This is obviously new functionality we did not discuss before, but it sounds very practical.

So two questions:
1) what do you think of standalone buy now outside of auctions?
2) what do you think of bidding outside of auctions like I described?

Reply:
> I see what you mean, it oculd make sense to have both together, but it does run the risk of the creator loosing out from bidding that would have exceeded the buy now price, but ok. However, I think one should definitively also be able to sell an NFT even when there is no auction ongoing, this is important when there is perhaps not a vibrant and liquid market and attention for a given NFT, or when you want some price discovery. This is not possible now?
> 
> It could even actually be add to this the ability for people to submit bids which are lower than the buy now price, so that the issuer gains some information about what people are willing to pay. If the issuer set the buy now price to the latest bid, then the transaction would happen. This is obviously new functionality we did not discuss before, but it sounds very practical.
> 
> So two questions:
> 
> 1. what do you think of standalone buy now outside of auctions?
> 2. what do you think of bidding outside of auctions like I described?
`I see what you mean, it oculd make sense to have both together, but it does run the risk of the creator loosing out from bidding that would have exceeded the buy now price, but ok`
I already foreseen this case. When bid exceeds `buy_now_price`, auction completes at the cost of `buy_now_price`, we do not withdraw more then necessary. )

Reply:
> I already foreseen this case. When bid exceeds buy_now_price, auction completes at the cost of buy_now_price, we do not withdraw more then necessary. )

Yes, but this does not solve the original problem, remember, the original problem I introduced was _creator loosing out from bidding that would have exceeded the buy now price_. 

I think for this reason we should basically keep _buy now_ as distinct from the normal time constrained auction. Remember that time constrained auctions are done precisely to generate urgency for the bidders to all come together and generate a high price.

> But the idea of bidding outside of auctions, you described here, looks very similar to auctions we have already, but with unlimited auction round duration.)

Almost, but not exactly. In a normal time constrained auction, the highest bidder above the minimum price will win the auction for sure at that price. In this kind of open auction, there is no minimum price and you are locked in for some period of time. After this time period, you can withdraw your bid and cash, but before, the owner can _accept_ your bid as the winner. This kind of acceptance by the owner does not exist in a normal auction.

Reply:
> I think for this reason we should basically keep _buy now_ as distinct from the normal time constrained auction. Remember that time constrained auctions are done precisely to generate urgency for the bidders to all come together and generate a high price.
> 
Yep, i misunderstood your point here. It makes sense, but since `buy now` is optional feature, it doesn't spoil classic auction flow. Auctioneer decides by himself whether he wants it or not. 
From my point of view, it can add some gamefication to regular auctions. Also it can help shorten overall auction time, which can be beneficial in some cases.

Reply:
What do you think about it?